source/common/tracing/http_tracer_impl.h

```diff
@@ -11,6 +11,13 @@
 #include "common/http/header_map_impl.h"
 #include "common/json/json_loader.h"
 
+#ifdef WIN32
+// <windows.h> defines macros for ERROR and TRUE, so we need to undef them in
+// case it's been included above
+#undef ERROR
+#undef TRUE
```

Comment:
Yeah, definitely not a fan of so many seemingly random places that we need to do these `#undef`. Would be very keen to see some way we can hide this and bound the number.

Reply:
Fantastic. In this case, I would suggest you formalize this rule in `tools/header_order.py`, so that we have a machine enforcement of this logic.

Reply:
Please review our submission of yesterday, localizing the decisions on posix / windows API headers and fix-ups of these to the include/envoy/common/platform.h file.

Expanding on that idea, it would also be possible to resolve the problems with Linux <sys/types.h> and #undef the 'major', 'minor' defines which cause significant noise at certain points in the build. It wouldn't be unreasonable to move this (basically portable) inclusion into platform.h as well.